File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var frequencySort = function(nums) {
    let obj = {}
    
    for(let i = 0; i < nums.length; i++){
        if(!obj[nums[i]]){
            obj[nums[i]] = 1
        } else{
            obj[nums[i]] ++
        }
    }
    nums.sort((a, b) => {
        
        if(obj[a] === obj[b]){
            return b - a
        }

        
        return obj[a] - obj[b]
    })
    return nums
};

File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.py
class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        obj = {}
        
        for x in range(len(nums)):
            if nums[x] not in obj:
                obj[nums[x]] = 1
            else:
                obj[nums[x]] += 1
        
        nums.sort(key=lambda x: (obj[x], -x))
        return nums